genesis/utils/ring_buffer.py

```diff
@@ -54,16 +54,23 @@ def append(self, tensor: torch.Tensor):
         self.buffer[self._idx_ptr.value].copy_(tensor)
         self._idx_ptr.value = (self._idx_ptr.value + 1) % self.N
 
-    def at(self, idx: int) -> torch.Tensor:
+    def at(self, idx: int | torch.Tensor) -> torch.Tensor:
         """
         Get a view of the tensor at the given index.
 
         Parameters
         ----------
-        idx : int
+        idx : int | torch.Tensor
             Index of the element to get, where 0 is the latest element, 1 is the second latest, etc.
+            Can be a tensor for batched indexing.
         """
-        return self.buffer[(self._idx_ptr.value - idx) % self.N]
+        rel_idx = (self._idx_ptr.value - idx) % self.N
+        if isinstance(idx, torch.Tensor):
+            batch_size = idx.shape[0] if len(idx.shape) > 0 else 1
+            env_indices = torch.arange(batch_size, device=idx.device)
+            return self.buffer[rel_idx, env_indices]
```

Comment:
What are you trying to achieve here? Why does the env indices always range from 0 to batch_size?

If you indices are a simple range, it would be more efficient to use `slice` instead of `torch.arange`. Moreover, since here it goes from 0 to batch_size, you should just do: `self.buffer[rel_idx, :batch_size]` no? Unless I'm missing something.